hedged_test.go

```diff
@@ -257,6 +257,128 @@ func TestBestResponse(t *testing.T) {
 	}
 }
 
+func TestOriginalResponseWins(t *testing.T) {
+	const shortest = 20 * time.Millisecond
+	timeouts := [...]time.Duration{shortest, 30 * shortest, 5 * shortest, shortest, shortest, shortest}
+	timeoutCh := make(chan time.Duration, len(timeouts))
+	for _, t := range timeouts {
+		timeoutCh <- t
+	}
+
+	url := testServerURL(t, func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(<-timeoutCh)
+	})
+
+	start := time.Now()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+	if err != nil {
+		t.Fatal(err)
+	}
+	client, metrics, err := hedgedhttp.NewClientAndStats(10*time.Millisecond, 5, nil)
+	if err != nil {
+		t.Fatalf("want nil, got %s", err)
+	}
+
+	checkAllMetricsAreZero(t, metrics)
+	resp, err := client.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer resp.Body.Close()
+
+	passed := time.Since(start)
+
+	if float64(passed) > float64(shortest)*2.5 {
+		t.Fatalf("want %v, got %v", shortest, passed)
+	}
+	if requestedRoundTrips := metrics.RequestedRoundTrips(); requestedRoundTrips != 1 {
```

Comment:
Nit: if we had a way to print all stats, a string comparison might be more readable